lib/live_ex.ex

```diff
@@ -26,32 +29,43 @@ defmodule LiveX do
       @doc """
       Dispatch an Action with a `type` and an optional payload.
       """
-      def dispatch(type, payload \\ %{}, socket) do
-        event = %{
-          type: String.to_atom(type),
-          payload: payload
-        }
+      def dispatch(type, payload \\ nil, socket)
+
+      @spec dispatch(atom, any, socket) :: map
+      def dispatch(type, payload, socket) when is_atom(type) do
+        action = %{type: type, payload: payload}
 
-        send(socket.assigns.pid, event)
+        send(socket.assigns.pid, action)
```

Comment:
This might be prone to cause race conditions. It would be better to use `handle_continue/2` callbacks to mutate the state (which does not work from your child components as your state lives in a different process)

Reply:
I see your point and I will address it once I move the Store to its own Process. However, this is not too big a problem right now as the Flux pattern does not require consistency, but only that state changes are sequential and go through a single Dispatcher.

Reply:
Addressed in #2